### PR TITLE
implement request_location_update via push notifications

### PR DESCRIPTION
**Pull request by Mic92**
I was not able to test this yet, since it seems that I cannot send pull notifications from a development app? https://github.com/home-assistant/home-assistant-android#setup-development-environment

### CODE REVIEW

File: app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt [hunk 1]
```diff
@@ -50,11 +51,23 @@ class MessagingService : FirebaseMessagingService() {
...
+            } else {
+                Log.d(TAG, "Message Notification: ${it.title} -> ${it.body}")
+                sendNotification(it.title, it.body!!)
+            }
         }
     }
 
+    private fun requestLocationUpdate() {
+        val intent = Intent(this, LocationBroadcastReceiver::class.java)
+        intent.action = LocationBroadcastReceiver.ACTION_REQUEST_LOCATION_UPDATES
```
Comment (JBassett) [inline app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt]: This intent action doesn't request a single high accuracy location update.  If you want that you will need a new action that calls `requestSingleAccurateLocation` and call that from here.

**Comment by Mic92** [comment]
Fixed

File: app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt [hunk 2]
```diff
@@ -50,11 +51,23 @@ class MessagingService : FirebaseMessagingService() {
         }
 
         remoteMessage.notification?.let {
-            Log.d(TAG, "Message Notification: ${it.title} -> ${it.body}")
-            sendNotification(it.title, it.body!!)
+            if (it.title == "request_location_update") {
```
Comment (JBassett) [inline app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt]: I just checked what the iOS app is doing, this should be message, not title.  That way we try and stay inline.

**Comment by Mic92** [comment]
All changes implemented.

**Comment by mdrichardson** [comment]
@Mic92 Nice work! Looks like we can do like iOS and just send `message: "request_location_update"` for this?

**Comment by JBassett** [comment]
After looking deeper, this is going to be affected by the same thing as https://github.com/home-assistant/home-assistant-android/issues/337.  I'll be making changes to fix both.

**Comment by Mic92** [comment]
@JBassett did you ever do this fix?

**Comment by JBassett** [comment]
Yes, this is working as expected.

**Pull request merged**